client/src/test/java/org/eclipse/hono/client/impl/DeviceConnectionClientImplTest.java

```diff
@@ -136,9 +143,73 @@ public void testSetLastKnownGatewayForDeviceSuccess(final VertxTestContext ctx)
         // WHEN setting the last known gateway
         client.setLastKnownGatewayForDevice("deviceId", "gatewayId", span.context())
                 .setHandler(ctx.succeeding(r -> {
-                    // THEN the response for setting the last known gateway has been handled by the service
-                    // and the span is finished
-                    verify(span).finish();
+                    ctx.verify(() -> {
+                        // THEN the response for setting the last known gateway has been handled by the service
+                        // and the span is finished
+                        verify(span).finish();
+                    });
+                    ctx.completeNow();
+                }));
+
+        final ArgumentCaptor<Message> messageCaptor = ArgumentCaptor.forClass(Message.class);
+        verify(sender).send(messageCaptor.capture(), any(Handler.class));
+        final Message response = ProtonHelper.message();
+        MessageHelper.addProperty(response, MessageHelper.APP_PROPERTY_STATUS, HttpURLConnection.HTTP_NO_CONTENT);
+        MessageHelper.addCacheDirective(response, CacheDirective.maxAgeDirective(60));
+        response.setCorrelationId(messageCaptor.getValue().getMessageId());
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+        client.handleResponse(delivery, response);
```

Comment:
this seems to be repeated over and over again, maybe you can extract it into a helper method?